src/main/java/edu/harvard/iq/dataverse/DataverseLocaleBean.java

```diff
@@ -0,0 +1,155 @@
+            locale= new Locale ("");
+        } else if(context.getViewRoot().getLocale().getLanguage()== "en_US" || context.getViewRoot().getLocale().getLanguage()== "en"){
+            locale= new Locale ("");
+        }else{
+            locale= context.getViewRoot().getLocale();
+        }
+    }
+
+
+    public DataverseLocaleBean(){
+
+    }
+
+    private static String readUrl(String urlString) throws Exception {
+        BufferedReader reader = null;
+        try {
+            URL url = new URL(urlString);
+            reader = new BufferedReader(new InputStreamReader(url.openStream()));
+            StringBuffer buffer = new StringBuffer();
+            int read;
+            char[] chars = new char[1024];
+            while ((read = reader.read(chars)) != -1)
+                buffer.append(chars, 0, read);
+
+            return buffer.toString();
+        } finally {
+            if (reader != null)
+                reader.close();
+        }
+    }
+
+
+    public List<DataverseLocale> getCountriesInMap() {
+        dataverseLocales = new ArrayList<DataverseLocale>();
+
+        try {
+            json_obj = new JSONObject(readUrl(json_url));
+            JSONArray json_array = json_obj.getJSONArray("languages");
+            for (int i = 0; i < json_array.length(); i++) {
+                String title = json_array.getJSONObject(i).getString("title");
+                String localeStr = json_array.getJSONObject(i).getString("locale");
+
+                dataverseLocales.add(new DataverseLocale(i , title, new Locale(localeStr)));
+            }
+        } catch (Exception e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+
+        return dataverseLocales;
+    }
+
+    public Locale getLocale(){
+        return this.locale;
+    }
+
+    public void setLocale(Locale _locale){
+        this.locale = _locale;
+    }
+
+    public String getLocaleCode() {
+        return localeCode;
+    }
+
+    public void setLocaleCode(String localeCode) {
+        this.localeCode = localeCode;
+    }
+    
+    public String getLocaleTitle() {
+        if (dataverseLocales != null) {
+            for (Iterator<DataverseLocale> it = dataverseLocales.iterator(); it.hasNext(); ) {
+                DataverseLocale dL = it.next();
+                if (dL.getId().toString().equals(localeCode)) {
+                    return dL.getDisplayName();
+                }
+            }
+        }
+        
+        return "English";
+    }
+
+    public void countryLocaleCodeChanged(String id){
+
+        for (Iterator<DataverseLocale> it = dataverseLocales.iterator(); it.hasNext(); ) {
+            DataverseLocale dL = it.next();
+
+            if (dL.getId().toString().equals(id)) {
+                FacesContext.getCurrentInstance()
+                        .getViewRoot().setLocale((Locale) dL.getName());
+
+                setLocaleCode(dL.getId().toString());
+                setLocale((Locale) dL.getName());
```

Comment:
should there be a `break` here? It would prevent extra loops if only one `DataverseLocal` can have the right `id`